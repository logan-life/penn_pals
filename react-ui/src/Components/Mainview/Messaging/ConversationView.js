import React, { useState, useEffect } from "react";
import Conversation from "./Conversation";
import Spinner from "react-bootstrap/Spinner";

export default function ConversationView({ sid, client, username }) {
  const [conversationObject, setConversationObject] = useState(null);

  useEffect(() => {
    client.getConversationBySid(sid).then((convo) => {
      setConversationObject(convo);
    });
  }, [client, sid]);

  let conversationContent;

  if (conversationObject) {
    conversationContent = (
      <Conversation
        conversationObject={conversationObject}
        myIdentity={username}
      />
    );
  } else {
    conversationContent = "";
  }

  return !conversationObject ? (
    <Spinner animation="border" role="status">
      <span className="sr-only">Loading Conversation</span>
    </Spinner>
  ) : (
    <div className="conversations-window-container">
      <div id="SelectedConversation">{conversationContent}</div>
    </div>
  );
}
