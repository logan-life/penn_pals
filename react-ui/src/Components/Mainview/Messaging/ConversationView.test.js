import ConversationView from './ConversationView';
const renderer = require('react-test-renderer');

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

// snapshot testing
test('ConversationView view matches snapshot', () => {
const component = renderer.create(<ConversationView />);
let tree = component.toJSON();
expect(tree).toMatchSnapshot();
});
