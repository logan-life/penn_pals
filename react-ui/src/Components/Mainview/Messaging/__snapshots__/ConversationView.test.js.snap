// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationView view matches snapshot 1`] = `
<div
  className="spinner-border"
  role="status"
>
  <span
    className="sr-only"
  >
    Loading Conversation
  </span>
</div>
`;
